### PR TITLE
W-16993325-pce4-upgrade-kt

### PR DESCRIPTION
**Pull request by kevintroller**
# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

File: modules/ROOT/pages/upgrade.adoc [hunk 1]
```diff
@@ -1,16 +1,126 @@
-= Upgrading Anypoint Platform PCE v3.2.3 ro v4.0.0
+= Upgrading Anypoint Platform PCE v3.2.3 to v4.0.0
 ifndef::env-site,env-github[]
 include::_attributes.adoc[]
 endif::[]
 
-== MuleSoft Professional Services Required
-Contact MuleSoft professional services prior to upgrading. MuleSoft professional services must be involved throughout the upgrade process. 
+[NOTE]
+If your Anypoint Platform PCE version is earlier than version 3.2.3, you must first upgrade to version 3.2.3 and then upgrade to version 4.0.
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/upgrade.adoc]:
```suggestion
If your Anypoint Platform PCE version is earlier than version 3.2.3, upgrade to version 3.2.3 first and then upgrade to version 4.0. 
```

File: modules/ROOT/pages/upgrade.adoc [hunk 2]
```diff
@@ -1,16 +1,126 @@
...
 ifndef::env-site,env-github[]
 include::_attributes.adoc[]
 endif::[]
 
-== MuleSoft Professional Services Required
-Contact MuleSoft professional services prior to upgrading. MuleSoft professional services must be involved throughout the upgrade process. 
+[NOTE]
+If your Anypoint Platform PCE version is earlier than version 3.2.3, you must first upgrade to version 3.2.3 and then upgrade to version 4.0. 
+
+Upgrading Anypoint Platform Private Cloud Edition version 3.2.3 to version 4.0, consist of the following process:
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/upgrade.adoc]:
```suggestion
To upgrade Anypoint Platform Private Cloud Edition version 3.2.3 to version 4.0:
```

File: modules/ROOT/pages/upgrade.adoc [hunk 3]
```diff
@@ -1,16 +1,126 @@
...
+
+[[update-dns]]
+== Update your DNS
+As an Anypoint Platform PCE user, you can access Anypoint Platform UI via DNS or directly via a load balancer.
+
+If you access via DNS, follow the steps to xref:upgrade.adoc#create-backup[create a backup].
+
+If you access via a load balancer, update your DNS following these steps: 
+
+. In Amazon Route53, create a new record to update the load balancer to point to a DNS, an example DNS for PCE 3.x is `pce3.mulesoft.com`.
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/upgrade.adoc]:
```suggestion
. In Amazon Route53, create a new record to update the load balancer to point to a DNS. An example DNS for PCE 3.x is `pce3.mulesoft.com`. 
```

File: modules/ROOT/pages/upgrade.adoc [hunk 4]
```diff
@@ -1,16 +1,126 @@
...
+. Access Anypoint Platform and select *Access Management*.
+. Select *DNS/IP*.
+. For *Platform DNS/IP*, enter the DNS address where you are hosting the platform.
++
+image::pce4-access-management.png["Platform DNS/IP field to enter DNS address in Access Management UI"]
++
+[start=5]
+. Click *Save*.
+. Access Anypoint Platform PCE UI via DNS to ensure the login is successful. 
+. In your Anypoint Platform PCE v3.2.3 cluster, scale down `nginx-wss-service`` to 0 replicas. After 5 minutes, scale it back to 3 replicas.
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/upgrade.adoc]:
```suggestion
. In your Anypoint Platform PCE v3.2.3 cluster, scale down `nginx-wss-service` to 0 replicas. After 5 minutes, scale it up to 3 replicas.
```

File: modules/ROOT/pages/upgrade.adoc [hunk 5]
```diff
@@ -1,16 +1,126 @@
...
+[[create-backup]]
+== Create a Backup
+To create a backup for upgrading Anypoint Platform PCE 3.2.3 to 4.0, follow these steps:
+
+. Run the following command to patch the Anypoint Platform PCE 3.2.4 cluster ConfigMap to include ignored secrets and buckets: 
++
+`kubectl apply -f backup-restore-config(1).yaml`
++
+[start=2]
+. Generate a backup using `gravity backup`:
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/upgrade.adoc]:
```suggestion
. Run the following command to generate a backup: 
```

File: modules/ROOT/pages/upgrade.adoc [hunk 6]
```diff
@@ -1,16 +1,126 @@
...
+mkdir nfs_mount_folder 
+mount ${nfsServer}/${nfsPath} nfs_mount_folder
+cp backup-3x-dd-mm-yy.tar.gz nfs_mount_folder/
+----
++
+Ensure the backup is copied to the NFS server so you can then perform a system restore.
 
-Before an Anypoint Platform Private Cloud Edition (Anypoint Platform PCE) upgrade is performed, you must verify that your working installation meets any additional requirements of the new Anypoint Platform PCE version, such as opening additional ports or adding a new port for the load balancer.
+== Install Anypoint Platform PCE v4.0
+After you complete the backup, the MuleSoft Professional Services team will assist you and provide you the steps to xref:install-workflow.adoc[install Anypoint Platform PCE v4.0].
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/upgrade.adoc]:
```suggestion
After you complete the backup, the MuleSoft Professional Services team will assist you with xref:install-workflow.adoc[installing Anypoint Platform PCE v4.0].
```

File: modules/ROOT/pages/upgrade.adoc [hunk 7]
```diff
@@ -1,16 +1,126 @@
...
 
 [NOTE]
-When upgrading from a previous Anypoint Platform PCE version, all policies available in the previous version remain. Policies are not replaced, but additional policies might be included with the upgrade.
+After the installation is complete don’t add data or new Mule apps in your Anypoint Platform PCE v4.0 environment.
+
+== Perform a System Restore
+To complete the upgrade, you must perform a system restore by following these steps:
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/upgrade.adoc]:
```suggestion
To complete the upgrade, perform a system restore by following these steps:
```

File: modules/ROOT/pages/upgrade.adoc [hunk 8]
```diff
@@ -1,16 +1,126 @@
...
+-H "Content-Type:application/json" \
+-d "{\"nfs-server\": \"${nfsServer}\", \
+\"nfs-path\": \"${nfsPath}\", \
+\"backup-file-name\": \"backup-3x-dd-mm-yy.tar.gz\" }"
+----
++
+Ensure the restore is successful by checking the restore logs. All the services and pods should be up and running successfully. 
+
+[start=3]
+. After the restore is complete from Anypoint Platform PCE v3.2.3 to v4.0 change back `skip-version-check` to `false`:
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/upgrade.adoc]:
```suggestion
. After the restore from Anypoint Platform PCE v3.2.3 to v4.0 is complete, change back `skip-version-check` to `false`:
```

File: modules/ROOT/pages/upgrade.adoc [hunk 9]
```diff
@@ -1,16 +1,126 @@
...
 
 [NOTE]
-When upgrading from a previous Anypoint Platform PCE version, all policies available in the previous version remain. Policies are not replaced, but additional policies might be included with the upgrade.
+After the installation is complete don’t add data or new Mule apps in your Anypoint Platform PCE v4.0 environment.
+
+== Perform a System Restore
+To complete the upgrade, you must perform a system restore by following these steps:
+
+. Temporarily set `skip-version-check` to `true` in your Anypoint Platform PCE v4.0 environment:
+.. Edit the ConfigMap by running `kubectl edit cm backup-restore-config`
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/upgrade.adoc]:
```suggestion
.. Edit the ConfigMap by running `kubectl edit cm backup-restore-config`.
```

File: modules/ROOT/pages/upgrade.adoc [hunk 10]
```diff
@@ -1,16 +1,126 @@
...
+-d "{\"nfs-server\": \"${nfsServer}\", \
+\"nfs-path\": \"${nfsPath}\", \
+\"backup-file-name\": \"backup-3x-dd-mm-yy.tar.gz\" }"
+----
++
+Ensure the restore is successful by checking the restore logs. All the services and pods should be up and running successfully. 
+
+[start=3]
+. After the restore is complete from Anypoint Platform PCE v3.2.3 to v4.0 change back `skip-version-check` to `false`:
+.. Edit the ConfigMap by running `kubectl edit cm backup-restore-config`
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/upgrade.adoc]:
```suggestion
.. Edit the ConfigMap by running `kubectl edit cm backup-restore-config`.
```

File: modules/ROOT/pages/upgrade.adoc [hunk 11]
```diff
@@ -1,16 +1,126 @@
...
+.. Set `skip-version-check` to `false`. +
+This ensures that there will be no blockling to a possible backup and restore process in the future.
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/upgrade.adoc]:
```suggestion
This ensures there's no blocker for a possible backup and restore process in the future. 
```

File: modules/ROOT/pages/upgrade.adoc [hunk 12]
```diff
@@ -1,16 +1,126 @@
...
+--data-raw '{"platform-dns":"test.upgrade1.pce.mulesoft.com"}'
+----
++
+[start=6]
+. In your Anypoint Platform PCE v3.2.3 cluster, scale down `nginx-wss-service` to 0 replicas. 
+. Ensure the login to Anypoint Platform PCE v4.0 works with the updated DNS.
+
+== Perform Post-Upgrade Configurations
+
+After you complete the upgrade, you can perform the following configurations:
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/upgrade.adoc]:
```suggestion
After you complete the upgrade, you can:
```

**Review by Cristian-Venticinque — APPROVED** [review]
Approved with comments

**Comment by fberckemeyer** [comment]
See also [my comment in Slack](https://salesforce-internal.slack.com/archives/C03E9379KSB/p1730154780869789?thread_ts=1730148375.547169&cid=C03E9379KSB).
IMHO, we not should make this information available publicly. 
The PCE runbook (https://pce-runbook.kbuild.msap.io/) might be a better home for this.

**Comment by kevintroller** [comment]
Based on PCE PM decision we'll not publish the guide externally.
See https://salesforce-internal.slack.com/archives/C03E9379KSB/p1730308516872069?thread_ts=1730148375.547169&cid=C03E9379KSB

**Pull request closed without merging**